devtools/server/actors/replay/connection.js

```diff
@@ -1080,10 +1093,27 @@ function collectUnresolvedSourceMapResources(mapText, mapURL, mapBaseURL) {
 }
 
 async function fetchText(url) {
+  let urlObj;
+  try {
+    urlObj = new URL(url);
+  } catch (err) {
+    urlObj = null;
+  }
+  if (urlObj && !["http:", "https:", "data:"].includes(urlObj.protocol)) {
+    urlObj = null;
+  }
+  if (!urlObj) {
+    console.warn("Unable to fetch recording resource", url);
+    return null;
+  }
+
   try {
     const response = await fetch(url);
     if (response.status < 200 || response.status >= 300) {
       console.error("Error fetching recording resource", url, response);
+      pingTelemetry("sourcemap-upload", "fetch-bad-status", {
+        status: response.status,
```

Comment:
Is there value in using a consistent payload here so we can more easily compare the failures? Could keep `status` but add `message` as well. I suppose you could `throw` the above instead and allow the `catch` to handle it consistently too.

Reply:
Good idea, thanks!